addons/mail_client_extension/controllers/mail_client_extension.py

```diff
@@ -14,32 +15,159 @@
+
+        return response
+
+    @http.route('/mail_client_extension/partners/search', type="json", auth="outlook", cors="*")
+    def res_partners_search(self, query, limit=30, **kwargs):
+        """
+        Used for the plugin search contact functionality where the user types a string query in order to search for
+        matching contacts, the string query can either be the name of the contact, it's reference or it's email.
+        We choose these fields because these are probably the most interesting fields that the user can perform a
+        search on.
+        The method returns an array containing the dicts of the matched contacts.
+        """
+
+        #In a multi-company environment, the method may return contacts not belonging to the company that the user
+        #is connected to, this may result in the user not being able to view the contact in Odoo, while this may happen
+        #it is not supported for now and users are encouraged to check if they are connected to the correct company before
+        # clicking on a contact.
+        filter_domain = ['|', '|', ('display_name', 'ilike', query), ('ref', '=', query), ('email', 'ilike', query)]
+        # Search for the partner based on the email.
+        # If multiple are found, take the first one.
+        partners = request.env['res.partner'].search(filter_domain, limit=limit)
+
+        partners = [
+            self._prepare_partner_values(partner)
+            for partner in partners
+        ]
+        return {"partners": partners}
+
+    @http.route('/mail_client_extension/partner/create', type="json", auth="outlook", cors="*")
+    def res_partner_create(self, email, name, company, **kwargs):
+        # TODO search the company again instead of relying on the one provided here?
+        # Create the partner if needed.
+        partner_info = {
+            'name': name,
+            'email': email,
+        }
+        if company and company > -1:
+            partner_info['parent_id'] = company
+        partner = request.env['res.partner'].create(partner_info)
+
+        response = {'id': partner.id}
+        return response
+
+    @http.route('/mail_client_extension/log_mail_content', type="json", auth="outlook", cors="*")
+    def log_mail_content(self, model, res_id, message):
+        if model not in self._mail_content_logging_models_whitelist():
+            raise Forbidden()
+        record = request.env[model].browse(res_id)
+        record.message_post(body=message)
+
     def _iap_enrich(self, domain):
         enriched_data = {}
         try:
-            response = request.env['iap.enrich.api']._request_enrich({domain: domain}) # The key doesn't matter
-        #except odoo.addons.iap.models.iap.InsufficientCreditError as ice:
+            response = request.env['iap.enrich.api']._request_enrich({domain: domain})  # The key doesn't matter
+        # except odoo.addons.iap.models.iap.InsufficientCreditError as ice:
```

Comment:
Is keeping this commented line interesting :D ?

Reply:
Nope, it's not :D